static/app/views/settings/project/projectSettingsLayout.tsx

```diff
@@ -1,48 +1,64 @@
-import {cloneElement, isValidElement} from 'react';
+import {Outlet, useOutletContext} from 'react-router-dom';
 
-import type {RouteComponentProps} from 'sentry/types/legacyReactRouter';
+import type {PlainRoute} from 'sentry/types/legacyReactRouter';
 import type {Project} from 'sentry/types/project';
 import useRouteAnalyticsParams from 'sentry/utils/routeAnalytics/useRouteAnalyticsParams';
-import useOrganization from 'sentry/utils/useOrganization';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useParams} from 'sentry/utils/useParams';
+import {useRoutes} from 'sentry/utils/useRoutes';
 import ProjectContext from 'sentry/views/projects/projectContext';
 import SettingsLayout from 'sentry/views/settings/components/settingsLayout';
 
-type Props = {
-  children: React.ReactNode;
-} & RouteComponentProps<{projectId: string}>;
+type InnerProps = {
+  params: {projectId: string};
+  project: Project;
+  routes: PlainRoute[];
+};
 
-type InnerProps = Props & {project: Project};
+type ProjectSettingsOutletContext = {
+  project: Project;
+};
+
+function ProjectSettingsOutlet(props: ProjectSettingsOutletContext) {
+  return <Outlet context={props} />;
+}
+
+export function useProjectSettingsOutlet() {
+  return useOutletContext<ProjectSettingsOutletContext>();
+}
+
+function InnerProjectSettingsLayout({params, routes, project}: InnerProps) {
+  const location = useLocation();
 
-function InnerProjectSettingsLayout({
-  params,
-  routes,
-  project,
-  children,
-  ...props
-}: InnerProps) {
   // set analytics params for route based analytics
   useRouteAnalyticsParams({
     project_id: project.id,
     project_platform: project.platform,
   });
 
-  const organization = useOrganization();
-
   return (
-    <SettingsLayout params={params} routes={routes} {...props}>
-      {children && isValidElement(children)
-        ? cloneElement<any>(children, {organization, project})
-        : children}
+    <SettingsLayout
+      params={params}
+      routes={routes}
+      location={location}
+      router={undefined as any}
+      route={undefined as any}
+      routeParams={undefined as any}
+    >
+      <ProjectSettingsOutlet project={project} />
     </SettingsLayout>
   );
 }
 
-function ProjectSettingsLayout({params, ...props}: Props) {
+export default function ProjectSettingsLayout() {
```

Comment:
`ProjectSettingsLayout`: created `ProjectSettingsOutlet` for passing project context, still passing router params to `SettingsLayout`